ReportUtilityBills/measure.rb

```diff
@@ -0,0 +1,792 @@
+    fuels[[FT::Elec, true]] = Fuel.new(meters: ["#{EPlus::FuelTypeElectricity}Produced:Facility"])
+    fuels[[FT::Gas, false]] = Fuel.new(meters: ["#{EPlus::FuelTypeNaturalGas}:Facility"])
+    fuels[[FT::Oil, false]] = Fuel.new(meters: ["#{EPlus::FuelTypeOil}:Facility"])
+    fuels[[FT::Propane, false]] = Fuel.new(meters: ["#{EPlus::FuelTypePropane}:Facility"])
+    fuels[[FT::WoodCord, false]] = Fuel.new(meters: ["#{EPlus::FuelTypeWoodCord}:Facility"])
+    fuels[[FT::WoodPellets, false]] = Fuel.new(meters: ["#{EPlus::FuelTypeWoodPellets}:Facility"])
+    fuels[[FT::Coal, false]] = Fuel.new(meters: ["#{EPlus::FuelTypeCoal}:Facility"])
+
+    fuels.each do |(fuel_type, is_production), fuel|
+      fuel.units = get_timeseries_units_from_fuel_type(fuel_type)
+    end
+
+    utility_rates = {}
+    utility_rates[FT::Elec] = UtilityRate.new
+    utility_rates[FT::Gas] = UtilityRate.new
+    utility_rates[FT::Oil] = UtilityRate.new
+    utility_rates[FT::Propane] = UtilityRate.new
+    utility_rates[FT::WoodCord] = UtilityRate.new
+    utility_rates[FT::WoodPellets] = UtilityRate.new
+    utility_rates[FT::Coal] = UtilityRate.new
+
+    utility_bills = {}
+    utility_bills[FT::Elec] = UtilityBill.new
+    utility_bills[FT::Gas] = UtilityBill.new
+    utility_bills[FT::Oil] = UtilityBill.new
+    utility_bills[FT::Propane] = UtilityBill.new
+    utility_bills[FT::WoodCord] = UtilityBill.new
+    utility_bills[FT::WoodPellets] = UtilityBill.new
+    utility_bills[FT::Coal] = UtilityBill.new
+
+    return fuels, utility_rates, utility_bills
+  end
+
+  def get_outputs(fuels)
+    fuels.each do |(fuel_type, is_production), fuel|
+      unit_conv = UnitConversions.convert(1.0, 'J', fuel.units)
+      unit_conv /= 139.0 if fuel_type == FT::Oil
+      unit_conv /= 91.6 if fuel_type == FT::Propane
+
+      fuel.timeseries = get_report_meter_data_timeseries(fuel.meters, unit_conv, 0)
+    end
+  end
+
+  def reporting_frequency_map
+    return {
+      'timestep' => 'Zone Timestep',
+      'hourly' => 'Hourly',
+      'daily' => 'Daily',
+      'monthly' => 'Monthly',
+    }
+  end
+
+  def get_report_meter_data_timeseries(meter_names, unit_conv, unit_adder)
+    return [0.0] * @timestamps.size if meter_names.empty?
+
+    vars = "'" + meter_names.uniq.join("','") + "'"
+    query = "SELECT SUM(VariableValue*#{unit_conv}+#{unit_adder}) FROM ReportMeterData WHERE ReportMeterDataDictionaryIndex IN (SELECT ReportMeterDataDictionaryIndex FROM ReportMeterDataDictionary WHERE VariableName IN (#{vars}) AND ReportingFrequency='#{reporting_frequency_map[timeseries_frequency]}' AND VariableUnits='J') GROUP BY TimeIndex ORDER BY TimeIndex"
+    values = @sqlFile.execAndReturnVectorOfDouble(query)
+    fail "Query error: #{query}" unless values.is_initialized
+
+    values = values.get
+    values += [0.0] * @timestamps.size if values.size == 0
+    return values
+  end
+
+  def get_auto_marginal_rate(runner, state_code, fuel_type)
```

Comment:
Since we're no longer adjusting the rates provided by EIA, we're essentially treating the rate as an _average_, not _marginal_, rate and then we're adding fixed charges to it. So I'm realizing that we are essentially double counting the fixed charges. 

So we either need to A) not used the fixed charge when auto is provided or B) go back to doing some kind of adjustment. Having a hard time choosing between the two, but leaning towards the latter...

Reply:
For B), here is where that adjustment was done previously: https://github.com/NREL/resstock/blob/v2.0.0/measures/UtilityBillCalculations/measure.rb#L378-L383.

Reply:
Thanks. Yes, let's go with that calculation again. I suggest creating a HouseholdConsumption.csv file that has Electricity and Natural Gas columns in it, and you can pull the data from the Electricity.csv and Natural gas.csv files in that measure.